.github/workflows/sonar.yml

```diff
@@ -67,77 +62,16 @@ jobs:
           path: ~/.sonar/cache
           key: ${{ runner.os }}-sonar
 
-      - name: Cache Frontend dependencies
-        uses: actions/cache@v3
-        with:
-          path: frontend/node_modules
-          key: ${{ runner.os }}-frontend-${{ hashFiles('frontend/package-lock.json')}}
-          restore-keys: ${{ runner.os }}-frontend
-
-      - name: Cache Giskard test resources
-        uses: actions/cache@v3
-        with:
-          path: ~/.giskard
-          key: ${{ runner.os }}-giskard-test-resources-${{ hashFiles('python-client/tests/fixtures/**/*py')}}
-          restore-keys: ${{ runner.os }}-giskard-test-resources
-
-      - name: Cache Python dependencies
-        uses: actions/cache@v3
-        with:
-          path: python-client/.venv
-          key: ${{ runner.os }}-${{ matrix.python-version }}-client-${{ hashFiles('python-client/pdm.lock')}}
-          restore-keys: ${{ runner.os }}-client
-
-      - uses: actions/setup-node@v3
-        with:
-          node-version: 16
-
       - name: Set up JDK 17
         uses: actions/setup-java@v3
```

Comment:
it might be faster to use `gradle-build-action` because it might cache gradle installation itself

Reply:
I will try that, but after this one I think.
I still have to work on CODEOWNER, gradle cache and some other stuff, but this PR is already a bit too big I think

Reply:
ok, could you create a separate card please?

Reply:
Done ! https://linear.app/giskard/issue/GSK-1717/use-gradle-build-action-instead-of-calling-gradlew